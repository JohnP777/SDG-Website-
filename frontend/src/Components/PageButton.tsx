import { Button } from '@mui/material';
import React from 'react';

export enum PageButtonColour {
    Green = '#9CCC92',
    Red = '#ed4431',
    White = 'white',
    Blue = '#4285F4'
}

const PageButton = ({ children, colour, textColour = 'white', startIcon, onClick }:
  {children?: React.ReactNode, colour: PageButtonColour, textColour?: string, startIcon?: React.ReactNode, onClick: () => void}) => {
  let variant: "contained" | "outlined" = "contained";
  if (textColour !== 'white') {
    variant = "outlined";
  }
  return (
    <Button variant={variant} onClick={onClick} sx={{
        background: colour,
        color: textColour,
        fontSize: '16px',
        borderRadius: '8px',
        textTransform: 'none',
        borderColor: 'black',
      }}
      startIcon={startIcon}
    >
      {children}
    </Button>
  );
}

export default PageButton;
