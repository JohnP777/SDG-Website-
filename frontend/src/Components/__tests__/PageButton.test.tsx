import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import PageButton, { PageButtonColour } from "../PageButton";

// Snapshot tests to check styling has not unintentionally changed since last time
it("snapshot test", () => {
  const onClick = jest.fn();
  const sampleText = 'Sample Text'
  const { asFragment } = render(<PageButton
    colour={PageButtonColour.Red}
    onClick={onClick}
  >{sampleText}</PageButton>);

  expect(asFragment()).toMatchSnapshot();
});

// DOM tests to check functionality and presence of elements
it("should call onClick when clicked on", () => {
  const onClick = jest.fn();
  const sampleText = 'Sample Text'
  render(<PageButton
    colour={PageButtonColour.Red}
    onClick={onClick}
  >{sampleText}</PageButton>);

  const button = screen.getByRole("button", { name: sampleText });

  expect(
    button,
  ).toBeInTheDocument();
  
  fireEvent.click(button);

  expect(onClick).toHaveBeenCalled();
});