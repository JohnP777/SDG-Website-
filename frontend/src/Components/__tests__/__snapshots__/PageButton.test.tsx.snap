// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-14281qt-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Sample Text
  </button>
</DocumentFragment>
`;
